### PR TITLE
tls: remove serving capability from deprecated apiserver cert, also fix Complete ClientCA bundle for apiserver

### PR DESCRIPTION
**Pull request by deads2k**
The kubelet terminates mTLS to determine the identity of the caller. This fixes the combined client bundle that can provide a stepping stone to getting the kubelet restricting properly.

@sjenning as we discussed yesterday
/assign @wking

### CODE REVIEW

File: pkg/asset/tls/apiserver.go [hunk 1]
```diff
@@ -31,27 +31,13 @@ func (a *APIServerCertKey) Dependencies() []asset.Asset {
...
-	}
+	dependencies.Get(kubeCA)
 
 	cfg := &CertCfg{
 		Subject:      pkix.Name{CommonName: "system:kube-apiserver", Organization: []string{"kube-master"}},
 		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
-		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 		Validity:     ValidityTenYears,
-		DNSNames: []string{
```
Comment (deads2k) [inline pkg/asset/tls/apiserver.go]: We've ratcheted through changes in kas-o, mco, kubelet, kas-o, and now we can fix this.  It's never used to serve anymore

File: pkg/asset/tls/apiserver.go [hunk 2]
```diff
@@ -538,12 +524,12 @@ var _ asset.Asset = (*KubeAPIServerCompleteClientCABundle)(nil)
 // Dependencies returns the dependency of the cert bundle.
 func (a *KubeAPIServerCompleteClientCABundle) Dependencies() []asset.Asset {
 	return []asset.Asset{
-		&KubeCA{},                              // TODO this should be removed once it never signs a client
-		&AdminKubeConfigCABundle{},             // admin.kubeconfig
-		&KubeletClientCABundle{},               // signed kubelet certs
-		&KubeControlPlaneCABundle{},            // controller-manager, scheduler
-		&KubeAPIServerToKubeletClientCertKey{}, // kube-apiserver to kubelet (kubelet piggy-backs on KAS client-ca)
```
Comment (deads2k) [inline pkg/asset/tls/apiserver.go]: This prevents kas-o from rotating client cert/key pairs.

File: pkg/asset/tls/apiserver.go [hunk 3]
```diff
@@ -538,12 +524,12 @@ var _ asset.Asset = (*KubeAPIServerCompleteClientCABundle)(nil)
 // Dependencies returns the dependency of the cert bundle.
 func (a *KubeAPIServerCompleteClientCABundle) Dependencies() []asset.Asset {
```
Comment (deads2k) [inline pkg/asset/tls/apiserver.go]: @sjenning this is the one it is legal to depend upon.

**Comment by abhinavdahiya** [comment]
This removes usages from a deprecated cert ?

**Comment by deads2k** [comment]
> This removes usages from a deprecated cert ?

Yeah, tightening things up so they can't grow back while we're working out full removal.

**Comment by wking** [comment]
Not very clear to me, but it's been open for a while, and CI is green, so:

/lgtm

**Comment by abhinavdahiya** [comment]
Just wanna take a quick look. Will hold cancel asap

File: pkg/asset/tls/apiserver.go [hunk 4]
```diff
@@ -31,27 +31,13 @@ func (a *APIServerCertKey) Dependencies() []asset.Asset {
 // Generate generates the cert/key pair based on its dependencies.
 func (a *APIServerCertKey) Generate(dependencies asset.Parents) error {
 	kubeCA := &KubeCA{}
-	installConfig := &installconfig.InstallConfig{}
```
Comment (abhinavdahiya) [inline pkg/asset/tls/apiserver.go]: you should also drop the dependency from https://github.com/openshift/installer/pull/1640/files#diff-c7316d3d0bee102140cf53dc3ae5edf9R27

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/installer/pull/1640#discussion_r276509444 is non-blocking.
i'll take a todo to fix this in another PR.

/hold cancel

**Pull request merged**